drift/core/telescope.py

```diff
@@ -918,6 +919,91 @@ def _beam(self, feed_ind, freq_ind):
 
         return beam
 
+    # ===================================================
+
+    # ====== Properties to help with draco pipeline =====
+
+    @cache.cached_property
+    def prodstack(self):
+        """Generate the results of a prodstack.
+
+        This is similar to the output of `uniquepairs`, but has the same typing as used
+        within draco.
+
+        Returns
+        -------
+        prodstack : np.ndarray
+            A structured array with (input_a, input_b) pairs.
+        """
+        upairs = self.uniquepairs
+
+        # Construct the return type using the same dtype length as used in the telescope
+        dtype = [("input_a", upairs.dtype), ("input_b", upairs.dtype)]
+
+        return upairs.ravel().view(dtype)
+
+    @cache.cached_property
+    def index_map_prod(self):
+        """Generate a *full triangle* `index_map/prod` like object.
+
+        Returns
+        -------
+        prodmap : np.ndarray
+            A structured array of (input_a, input_b) pairs for the upper triangle.
+        """
+        tpairs = np.array(np.triu_indices(self.nfeed))
+        dtype = [("input_a", tpairs.dtype), ("input_b", tpairs.dtype)]
+
+        return tpairs.T.flatten().view(dtype)
+
+    @cache.cached_property
+    def index_map_stack(self):
+        """Generate an `index_map/stack` like object.
+
+        Returns
+        -------
+        stack : np.ndarray
+            A structured array with (prod_ind, conj) pairs the same length as
+            `unique_pairs`.
+        """
+        # Taken from draco.util.tools.cmap, but we can't depend on it in driftscan
```

Comment:
Should we consider moving `cmap()` and `icmap()` from `draco.utils.tools` into `drift.core.telescope` or elsewhere in `driftscan`? I think this would be preferable to having two copies of the same routine. (`cmap()` only appears to be used in `draco.analysis.transform.CollateProducts`)

Reply:
The problem with cmap is that it's worked it's way into a bunch of people's random scripts over the years. I guess we would need to patch it back into draco.util.tools. Overall I think we need to have a bit of a figuring out of where to put miscellaneous utility functions that cross a lot of boundaries like this. So with that in mind, I'm going to punt on it.

Reply:
chime-experiment/Pipeline#102